compiler/src/dotty/tools/dotc/printing/PlainPrinter.scala

```diff
@@ -68,6 +68,15 @@ class PlainPrinter(_ctx: Context) extends Printer {
       }
     else tp
 
+  private def sameBound(lo: Type, hi: Type): Boolean =
+    try lo =:= hi
+    catch { case ex: Throwable => false }
+
+  private def homogenizeArg(tp: Type) = tp match {
```

Comment:
In fact, the name was correct, but the condition to do this only under `homogenizedView` was lacking.